### PR TITLE
Use separate file-events library

### PR DESCRIPTION
**Pull request by wolfs**
To see if the Gradle build can consume the separate `file-events` library.

### CODE REVIEW

**Comment by wolfs** [comment]
Looks like we need to load the file events library separately, so it ends up in the right directory.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java [hunk 1]
```diff
@@ -109,7 +113,16 @@ public static synchronized void initialize(File userHomeDir, boolean initializeJ
...
+                    if (useNativeIntegrations && initializeAdditionalNativeLibraries) {
+                        useFileSystemWatching = true;
+                        try {
+                            FileEvents.init(nativeBaseDir);
+                        } catch (NativeIntegrationUnavailableException ex) {
+                            LOGGER.info("Native file system watching is not available for this operating system.", ex);
+                            useFileSystemWatching = false;
+                        }
+                    }
+                    if (initializeAdditionalNativeLibraries) {
```
Comment (lptr) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java:125]: I think things would be more readable if this `if` was wrapping the one currently right above it.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java [hunk 2]
```diff
@@ -297,6 +310,11 @@ protected NativeCapabilities createNativeCapabilities() {
             public boolean isNativeIntegrationAvailable() {
                 return useNativeIntegrations;
             }
+
+            @Override
+            public boolean isFileSystemWatchingAvailable() {
+                return useFileSystemWatching;
```
Comment (lptr) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java:316]: Shall we name the local variable and the getter the same?

File: subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy [hunk 3]
```diff
@@ -37,7 +37,7 @@ import static org.hamcrest.MatcherAssert.assertThat
 
 abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
 
-    protected static final THIRD_PARTY_LIB_COUNT = 132
+    protected static final THIRD_PARTY_LIB_COUNT = 140
```
Comment (lptr) [inline subprojects/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:40]: Wow. We really should start merging libraries.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java (hunk 1 above)
Comment (wolfs) [reply subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java:116]: Done.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java (hunk 2 above)
Comment (wolfs) [reply subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java:316]: I changed the name of the `getter`.

**Comment by wolfs** [comment]
@lptr PTAL!

**Pull request merged**